irs/core/upload_handler.py

```diff
@@ -28,121 +29,150 @@
     retrieve_part,
     retrieve_parts,
 )
-from irs.adapters.outbound.kafka_producer import (
-    produce_failure_event,
-    produce_success_event,
-)
 from irs.config import CONFIG
+from irs.ports.inbound.upload_handler import UploadHandlerPort
+from irs.ports.outbound.event_pub import EventPublisherPort
 
 
-async def process_new_upload(  # pylint: disable=too-many-locals
-    *, object_id: str, object_size: int, public_key: str, checksum: str
-):
-    """
-    Forwards first file part to encryption key store, retrieves file encryption
-    secret(s) (K_data), decrypts file and computes checksums
-    """
-    part_size = calc_part_size(file_size=object_size)
-    try:
-        download_url = await get_download_url(object_id=object_id)
-        part = await retrieve_part(url=download_url, start=0, stop=part_size - 1)
-        secret, secret_id, offset = call_eks_api(
-            file_part=part, public_key=public_key, api_url=CONFIG.eks_url
-        )
-        (
-            part_checksums_md5,
-            part_checksums_sha256,
-            content_checksum_sha256,
-        ) = await compute_checksums(
-            download_url=download_url,
-            secret=secret,
+class UploadHandler(UploadHandlerPort):
+    """A service that validates uploaded files"""
```

Comment:
Maybe define more from the perspective of this service and avoid assumptions about other services.
```suggestion
class Interrogator(InterrogatorPort):
    """A service that validates the content of encrypted files and performs a re-encryption."""
```